crypto/pkcs7/pk7_doit.c

```diff
@@ -1117,6 +1120,14 @@ int PKCS7_signatureVerify(BIO *bio, PKCS7 *p7, PKCS7_SIGNER_INFO *si,
         goto err;
     }
 
+    if (OBJ_obj2nid(si->digest_enc_alg->algorithm) == EVP_PKEY_RSA_PSS) {
+        i = ossl_rsa_pss_to_ctx(ctx, NULL, si->digest_enc_alg, pkey);
+        if (i <= 0) {
+            ret = -1;
+            goto err;
+        }
```

Comment:
Hmm, I think if this works, it works only by accident as EVP_DigestVerify should be used.